anynote/api/views.py

```diff
@@ -23,29 +32,108 @@ class UsersApiView(generics.ListCreateAPIView):
     queryset = User.objects.all()
     serializer_class = UsersSerializer
     authentication_classes = [JWTAuthentication]
-    permission_classes = [permissions.IsAuthenticated]
+    permission_classes = [permissions.IsAuthenticated, permissions.IsAdminUser]
+
+    # def post(self, request, *args, **kwargs):
+    #     serializer = UsersSerializer(data=request.data)
+    #     serializer.is_valid(raise_exception=True)
+    #     serializer.save()
+    #
+    #     return Response(serializer.data)
 
 
 usersApiView = UsersApiView.as_view()
 
 
 class UserApiView(generics.RetrieveUpdateDestroyAPIView):
     """
-    API endpoint that allows to get a specific ser data.
+    API endpoint that allows to get a specific user data.
 
     Only for debugging
     permission: is_staff
 
     """
     queryset = User.objects.all()
-    lookup_field = "pk"
     serializer_class = UsersSerializer
     authentication_classes = [JWTAuthentication]
     permission_classes = [permissions.IsAuthenticated]
 
+    def get_object(self):
+        access_token = self.request.META.get('HTTP_AUTHORIZATION').split()[1]
+        print(access_token)
+        try:
+            decoded_token = jwt.decode(access_token, settings.SECRET_KEY, algorithms=['HS256'])
+            user_id = decoded_token.get('user_id')
+            if user_id:
+                return User.objects.get(pk=user_id)
+            else:
+                raise Http404("Note not found.")
+
+        except jwt.exceptions.InvalidSignatureError:
+            raise PermissionDenied("Invalid access token.")
+
+        except NoteModel.DoesNotExist:
+            raise Http404("Note not found.")
```

Comment:
As we discussed, rename this UserApiView to AccountApiView and return old UserApiView from older commit. The endpoint users/<int:pk> will be used for testing purposes.